cmd/keymasterd/idp_oidc.go

```diff
@@ -324,21 +324,31 @@ func (state *RuntimeState) idpOpenIDCTokenHandler(w http.ResponseWriter, r *http
 	//formClientID := r.Form.Get("clientID")
 	logger.Printf("%+v", r)
 
+	unescapeAuthCredentials := true
 	clientID, pass, ok := r.BasicAuth()
 	if !ok {
-		state.writeFailureResponse(w, r, http.StatusUnauthorized, "")
-		return
+		logger.Printf("warn: basic auth Missing")
+		clientID = r.Form.Get("client_id")
+		pass = r.Form.Get("client_secret")
+		if len(clientID) < 1 || len(pass) < 1 {
+			logger.Printf("Cannot get auth credentials in auth request")
+			state.writeFailureResponse(w, r, http.StatusUnauthorized, "")
+			return
+		}
+		unescapeAuthCredentials = false
 	}
 	// https://tools.ietf.org/html/rfc6749#section-2.3.1 says the client id and password
 	// are actually url-encoded
 	//logger.Debugf(3, "Pre escaped auth: username=%s, pass=%s", clientID, pass)
-	unescapedClientID, err := url.QueryUnescape(clientID)
-	if err == nil {
-		clientID = unescapedClientID
-	}
-	unescapedPass, err := url.QueryUnescape(pass)
-	if err == nil {
-		pass = unescapedPass
+	if unescapeAuthCredentials {
+		unescapedClientID, err := url.QueryUnescape(clientID)
+		if err == nil {
+			clientID = unescapedClientID
+		}
+		unescapedPass, err := url.QueryUnescape(pass)
+		if err == nil {
+			pass = unescapedPass
+		}
 	}
 	//logger.Debugf(3, "username=%s, pass=%s", clientID, pass)
```

Comment:
Please either uncomment or delete. There should be no commented out Debugf calls.